sdk/resourcemanager/azure-resourcemanager-privatedns/src/main/java/com/azure/resourcemanager/privatedns/implementation/PrivateDnsZoneImpl.java

```diff
@@ -394,6 +411,7 @@ public Update withETagCheck(String etagValue) {
             .map(innerToFluentMap(this))
             .map(privateDnsZone -> {
                 etagState.clear();
+                tagsCopy = serializeTags(privateDnsZone.tags());
```

Comment:
I would suggest you do this on `update`. Else I won't know how many places this line is needed (e.g. if I add a `updateResourceAsync` method later, how do I know I need to do this?).

Reply:
Same concern would apply to ETagState too?

Reply:
ETagState might be a bit different? As it is used both in create or update.

But the `tagsCopy` is only for update flow, hence it just need to be done in the entry point of update flow, which is `update` method.